### ISSUE TITLE
Panic on google_sql interface {} is nil, not map[string]interface {}

### ISSUE THREAD

**Issue by meyskens** [terraform-providers/terraform-provider-google#2220]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment
* If an issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to "hashibot", a community member has claimed the issue already.

### Terraform Version

```
Terraform v0.11.8
+ provider.google v1.19.0
+ provider.google-beta v1.19.0
```

### Affected Resource(s)

* google_sql_user
* google_sql_database_instance

### Terraform Configuration Files

```tf
module "lookout_staging_cloudsql" {
  source      = "../../modules/google-cloud/sql" 
  project = "srcd-public-development"
  name = "lookout-staging"
  machine_type = "db-f1-micro"
  user_name = "maartje"
  user_host = "*"
  user_password = "<redacted>"
  db_name = "lookout"
}
```

```tf
resource "google_sql_database_instance" "primary" {
  name             = "${var.name}-primary"
  region           = "${var.region}"
  project          = "${var.project}"
  database_version = "POSTGRES_9_6"

  settings {
    tier              = "${var.machine_type}"
    disk_type         = "${var.disk_type}"
    disk_size         = "${var.disk_size}"
    disk_autoresize   = "${var.disk_autoresize}"
    activation_policy = "ALWAYS"
    availability_type = "REGIONAL"
    replication_type  = "${var.replication_type}"

    backup_configuration {
      enabled = true
    }
  }
}

resource "google_sql_database_instance" "secondary" {
  name                 = "${var.name}-secondary"
  region               = "${var.region}"
  project              = "${var.project}"
  database_version     = "POSTGRES_9_6"
  master_instance_name = "${google_sql_database_instance.primary.name}"

  settings {
    tier              = "${var.machine_type}"
    disk_type         = "${var.disk_type}"
    disk_size         = "${var.disk_size}"
    disk_autoresize   = "${var.disk_autoresize}"
    activation_policy = "ALWAYS"
    availability_type = "REGIONAL"
    replication_type  = "${var.replication_type}"
  }
}

resource "google_sql_user" "default" {
  name     = "${var.user_name}"
  project  = "${var.project}"
  instance = "${google_sql_database_instance.primary.name}"
  host     = "${var.user_host}"
  password = "${var.user_password}"
}

resource "google_sql_database" "default" {
  name     = "${var.db_name}"
  project  = "${var.project}"
  instance = "${google_sql_database_instance.primary.name}"
  charset  = "UTF8"   
}

```

### Panic Output

https://gist.github.com/meyskens/be4194a44bad433060bfad952e8708ec

### Expected Behavior

Should have created the resources.

### Actual Behavior

Panics after creating the primary CloudSQL instance.

### Steps to Reproduce

1. `terraform plan -out plan.out`
1. `terraform apply plan.out`

**Comment by rileykarson** [comment]
I'm having trouble reproducing this - would you be able to post a minimal config?

When I try something like

```hcl
resource "google_sql_database_instance" "primary" {
  provider         = "google-beta"
  name             = "somename-primary"
  database_version = "POSTGRES_9_6"
  region           = "us-central1"

  settings {
    tier              = "db-f1-micro"
    disk_type         = "PD_SSD"
    disk_size         = "50"
    disk_autoresize   = "false"
    activation_policy = "ALWAYS"
    availability_type = "REGIONAL"
    replication_type  = "SYNCHRONOUS"

    backup_configuration {
      enabled = true
    }
  }
}

resource "google_sql_database_instance" "secondary" {
  provider             = "google-beta"
  name                 = "somename-secondary"
  database_version     = "POSTGRES_9_6"
  master_instance_name = "${google_sql_database_instance.primary.name}"
  region               = "us-central1"

  settings {
    tier              = "db-f1-micro"
    disk_type         = "PD_SSD"
    disk_size         = "50"
    disk_autoresize   = "false"
    activation_policy = "ALWAYS"
    replication_type  = "SYNCHRONOUS"
  }
}

provider "google-beta" {
}
```

I don't get the error.

**Comment by meyskens** [comment]
@rileykarson thanks for the fast reply. I did just try to replicate it in test config to report it here but I did not get the error anymore. Will continue to investigate this.

**Comment by meyskens** [comment]
I think i found the problem here. In my original code I kept `replica_configuration` empty as I had commented something out. Which to keep it clean in this ticked I deleted. I removed and it is creating now.

```
replica_configuration {
  // commented code
}
```

**Comment by paddycarver** [comment]
I believe this should be fixed with the release of v1.19.1. Please update your provider, and feel free to comment if you continue to experience problems.
